packages/core/src/browser/terminal/terminal-service.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+import { TerminalWidgetOptions, TerminalWidget } from "./terminal-model";
+
+/**
+ * Service for manipulation terminal widgets.
+ */
+export const TerminalService = Symbol("TerminalService");
+export interface TerminalService {
+    /**
+     * Create new terminal with predefined options;
+     * @param options - terminal options.
+     */
+    newTerminal(options: TerminalWidgetOptions): Promise<TerminalWidget>;
+
+    /**
+     * Display new terminal widget on the bottom panel.
+     * @param termWidget - widget to attach to the bottom panel.
+     */
+    activateWidget(termWidget: TerminalWidget): void;
+
+    /**
+     * Hide panel where is located created terminal widget.
+     * @param termWidget - terminal widget to hide.
+     */
+    collapseWidget(termWidget: TerminalWidget): void;
```

Comment:
Probably it is just me, but these methods are strange here. At least the naming. If this is the terminal service API, why do not we have a `closeWidget` too?

Reply:
I think method "create" is not strange. With help this method you can create terminal widget. I did not apply method "close", because terminal widget has method dispose(). But it's not a problem to apply method "close". About "activateWidget", "collapseWidget", yes maybe we don't need it in the Theia core. It was created for plugin api... If you are against this methods here I can implement them in the plugin model (it is not merged yet and during development located in the wiptheia repo).

Reply:
> because terminal widget has method dispose().

I see. It makes sense.